### PR TITLE
feat: inso collection runner

### PR DESCRIPTION
**Pull request by jackkav**
- [x] works `inso run collection`
- [x] logging
- [x] request list by workspaceId should run all requests not just top level requests
- [x] fixtures should point at local server

future work
- example fixtures of bugs and things to build
- request group selection
- scripting

thoughts
- it would be nice to add a folder selection step but asks a lot of traversal problems so will scope it out

### CODE REVIEW

File: packages/insomnia-inso/src/cli.ts [hunk 1]
```diff
@@ -295,6 +296,100 @@ export const go = (args?: string[]) => {
...
+    .action(async (identifier, cmd: { env: string; disableCertValidation: true; requestNamePattern: string; bail: boolean }) => {
+      const globals: { config: string; workingDir: string; exportFile: string; ci: boolean; printOptions: boolean; verbose: boolean } = program.optsWithGlobals();
+
+      const commandOptions = { ...globals, ...cmd };
+      const __configFile = await loadCosmiConfig(commandOptions.config, commandOptions.workingDir);
+
+      const options = {
+        reporter: defaultReporter,
+        ...__configFile?.options || {},
+        ...commandOptions,
```
Comment (ihexxa) [inline packages/insomnia-inso/src/cli.ts:314]: nit: just double check that command options will be overridden by the config file? Original I thought command options might have highest priority.

File: packages/insomnia-inso/src/cli.ts [hunk 2]
```diff
@@ -295,6 +296,100 @@ export const go = (args?: string[]) => {
...
+        pathToSearch = localAppDir;
+      } else {
+        pathToSearch = path.resolve(options.workingDir || process.cwd(), options.exportFile || '');
+      }
+      if (options.reporter && !reporterTypesSet.has(options.reporter)) {
+        logger.fatal(`Reporter "${options.reporter}" not unrecognized. Options are [${reporterTypes.join(', ')}].`);
+        return process.exit(1);
+      }
+
+      const db = await loadDb({
```
Comment (ihexxa) [inline packages/insomnia-inso/src/cli.ts:333]: Probably we could be a bit defensive here and check `loadDb` failure.

File: packages/insomnia-inso/src/cli.ts [hunk 3]
```diff
@@ -295,6 +296,100 @@ export const go = (args?: string[]) => {
...
+        const sendRequest = await getSendRequestCallbackMemDb(environment._id, db, { validateSSL: !options.disableCertValidation });
+        let success = true;
+        for (const req of requests) {
+          if (options.bail && !success) {
+            return;
+          }
+          logger.log(`Running request: ${req.name} ${req._id}`);
+          const res = await sendRequest(req._id);
+          logger.trace(res);
+          if (res.status !== 200) {
```
Comment (ihexxa) [inline packages/insomnia-inso/src/cli.ts:381]: nit: Just a todo item that we might check test results to decide if it is failed in the future.

**Review by ihexxa — APPROVED** [review]
Just added some minor comments.

File: packages/insomnia-inso/src/cli.ts (hunk 2 above)
Comment (jackkav) [reply packages/insomnia-inso/src/cli.ts:333]: I agree the db load logic is over complex I'm thinking of a better way to abstract it. Will follow up with a pr soon

**Pull request merged**